### PR TITLE
Code refactorization in compartment and VSD loader.

### PR DESCRIPTION
**Pull request by hernando**
Axon sections are considered again by the compartment loader and VSD
loaders. Test results had to be adjusted accordingly. The results
have been compared against the old code with a hack to include
axon sections and they match.

### CODE REVIEW

File: fivox/helpers.h [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+#include <brion/compartmentReport.h>
+
+#include <lunchbox/log.h>
+
+namespace fivox
+{
+namespace helpers
+{
+
+/** Typle of buffer offset, cell index, section ID, compartment counts */
```
Comment (tribal-tec) [inline fivox/helpers.h]: typo Typle

File: fivox/helpers.h [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+typedef std::tuple< size_t, uint32_t, uint32_t, uint16_t >  MappingElement;
+typedef std::vector< MappingElement > FlatInverseMapping;
+
+/**
+ * Computes a report mapping that for enumerates compartments in the same
```
Comment (tribal-tec) [inline fivox/helpers.h]: that for enumerates??

File: fivox/helpers.h [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+ * order they appear in the data buffer.
+ * @param report The report from which the mapping is computed.
+ * @return A list of typles (offset, cell index, section ID, compartment count).
```
Comment (tribal-tec) [inline fivox/helpers.h]: tuples

File: fivox/helpers.h [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+ * @param The report from which the compartments per section are obtained
+ * @param output The output event source. Events are added in the morphology
+ *        iteration order, starting with the soma and then all the dendrites.
+ */
+inline void addCompartmentEvents(
+    const brain::neuron::Morphologies& morphologies,
+    const brion::CompartmentReport& report, EventSource& output )
+{
+    auto mapping = computeInverseMapping( report );
+    for( auto i : mapping )
```
Comment (tribal-tec) [inline fivox/helpers.h]: 'i' can be const&?

File: fivox/helpers.h [hunk 5]
```diff
@@ -0,0 +1,129 @@
...
+ *        mapping.
+ * @param The report from which the compartments per section are obtained
+ * @param output The output event source. Events are added in the morphology
+ *        iteration order, starting with the soma and then all the dendrites.
+ */
+inline void addCompartmentEvents(
+    const brain::neuron::Morphologies& morphologies,
+    const brion::CompartmentReport& report, EventSource& output )
+{
+    auto mapping = computeInverseMapping( report );
```
Comment (tribal-tec) [inline fivox/helpers.h]: const& ?

**Comment by hernando** [comment]
Retest this please

**Comment by hernando** [comment]
Retest this please

**Comment by jafyvilla** [comment]
Retest this please

**Comment by jafyvilla** [comment]
:+1:

**Pull request merged**